src/source/tile.js

```diff
@@ -413,17 +413,19 @@ class Tile {
     }
 
     setFeatureState(states: LayerFeatureStates) {
-        if (!this.latestRawTileData || Object.keys(states).length === 0) return;
-
-        if (!this.vtLayers) {
-            this.vtLayers = new vt.VectorTile(new Protobuf(this.latestRawTileData)).layers;
+        if (!this.latestFeatureIndex ||
+            !this.latestRawTileData ||
```

Comment:
Does it need to check for `this.latestRawTileData` since `this.latestFeatureIndex` has its own copy of the raw tile data

Reply:
Ah good point. It should check `this.latestFeatureIndex.rawTileData`. They'll have the same value because of the logic in `loadVectorData`, but the state of `FeatureIndex` is what we care about directly. Also, the tile workers should never generate a `FeatureIndex` that doesn't have an accompanying `rawTileData` (unless they're doing an update)... but since that's outside of the control of `Tile`, I think we still want to maintain the check.